mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageIdMapperTest.java

```diff
@@ -331,6 +331,19 @@ public void setFlagsShouldReturnUpdatedFlagsWhenMessageIsInOneMailbox() throws E
     }
 
     @ContractTest
+    public void setFlagsShouldReturnEmptyWhenMailboxIdsIsEmpty() throws Exception {
+        message1.setUid(mapperProvider.generateMessageUid());
+        message1.setModSeq(mapperProvider.generateModSeq(benwaInboxMailbox));
+        sut.save(message1);
+
+        MessageId messageId = message1.getMessageId();
+        Flags newFlags = new Flags(Flag.ANSWERED);
+        Map<MailboxId, UpdatedFlags> flags = sut.setFlags(messageId, ImmutableList.<MailboxId> of(), newFlags, MessageManager.FlagsUpdateMode.REMOVE);
+
+        assertThat(flags).isEmpty();
```

Comment:
Can you check that the original message has not been modified too please?
And can you add some tests with several mailboxes?